### PR TITLE
Change Rate Behavior

### PR DESCRIPTION
**Pull request by ajholland**
As a followup to #525 this sets the default rate to 10,000 pps. 

Scan at full line rate by passing `-r 0` 

Behaviors changed: 
-  Default is now 10,000 pps. It was previously full line rate. 
-  Setting 0 as the rate will scan at full line rate.

### CODE REVIEW

**Comment by zakird** [comment]
I think it probably makes more sense to have this be a default within ZMap and not just something in the configuration file. I'd like ZMap to act non-broken without a conf.

**Comment by ajholland** [comment]
Fair enough! I'll work on that.

**Comment by ajholland** [comment]
Changed it, wasn't sure if it should have a comment to not be a magic number, but it seemed redundant since it's already the default struct. Wasn't sure if you'd rather have it set as a capitalized const somewhere? maybe just in this file?

**Comment by zakird** [comment]
I think it's probably fine there. We have a bunch of constants just set as defaults there.

File: src/send.c [hunk 1]
```diff
@@ -174,7 +174,7 @@ iterator_t *send_init(void)
 		    zconf.bandwidth, pkt_len / 8, zconf.rate);
 	}
 	// log rate, if explicitly specified
-	if (zconf.rate <= 0) {
+	if (zconf.rate < 0) {
```
Comment (zakird) [inline src/send.c]: Is rate ever below zero, or was there some weird casting going on where values < 0.5 were being rounded down to zero?

Comment (zakird) [reply src/send.c]: It looks `zconf.rate` is an integer.

Comment (zakird) [reply src/send.c]: Looking at L160 above it looks like this should be fine. @ajholland does this also look alright to you?

**Comment by zakird** [comment]
approved. if that comment looks reasonable @ajholland, let's merge it!

File: src/send.c (hunk 1 above)
Comment (ajholland) [reply src/send.c]: Lgtm!

**Comment by ajholland** [comment]
Lgtm!

**Pull request merged**